contracts/protocol/pool/PoolMarketplace.sol

```diff
@@ -122,21 +135,43 @@ contract PoolMarketplace is
                 .getMarketplace(marketplaceId);
             DataTypes.OrderInfo memory orderInfo = IMarketplace(
                 marketplace.adapter
-            ).getAskOrderInfo(payload, WETH);
-            orderInfo.taker = onBehalfOf;
+            ).getAskOrderInfo(payload, weth);
+            orderInfo.taker = msg.sender;
+
+            // Once we encounter a listing using WETH, then we convert all our ethLeft to WETH
+            // this also means that the parameters order is very important
+            //
+            // frontend/sdk needs to guarantee that WETH orders will always be put after ALL
```

Comment:
how about add an explicit check if WETH orders have been put after ETH orders?

Reply:
feel it requires looping through the orders